File: src/main/java/org/taktik/icure/dao/EntityTemplateDAO.java
/*
 * Copyright (C) 2018 Taktik SA
 *
 * This file is part of iCureBackend.
 *
 * iCureBackend is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License version 2 as published by
 * the Free Software Foundation.
 *
 * iCureBackend is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with iCureBackend.  If not, see <http://www.gnu.org/licenses/>.
 */

package org.taktik.icure.dao;

import java.util.List;

import org.ektorp.support.View;
import org.taktik.icure.entities.EntityTemplate;

public interface EntityTemplateDAO extends GenericDAO<EntityTemplate> {

	@View(name = "by_user_type_descr", map = "classpath:js/patient/By_hcparty_contains_name_map.js")
	List<EntityTemplate> getByUserIdTypeDescr(String userId, String type, String searchString, Boolean includeEntities);

	@View(name = "by_type_descr", map = "classpath:js/patient/By_hcparty_contains_name_map.js")
	List<EntityTemplate> getByTypeDescr(String type, String searchString, Boolean includeEntities);
}

File: src/main/kotlin/org/taktik/icure/asyncdao/EntityTemplateDAO.kt
/*
 *  iCure Data Stack. Copyright (c) 2020 Taktik SA
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU Affero General Public License as
 *     published by the Free Software Foundation, either version 3 of the
 *     License, or (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful, but
 *     WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *     Affero General Public License for more details.
 *
 *     You should have received a copy of the GNU Affero General Public
 *     License along with this program.  If not, see
 *     <https://www.gnu.org/licenses/>.
 */

package org.taktik.icure.asyncdao

import kotlinx.coroutines.flow.Flow
import org.taktik.icure.entities.EntityTemplate

interface EntityTemplateDAO: GenericDAO<EntityTemplate> {
    suspend fun getByUserIdTypeDescr(userId: String, type: String, searchString: String?, includeEntities: Boolean?): Flow<EntityTemplate>
    suspend fun getByTypeDescr(type: String, searchString: String?, includeEntities: Boolean?): Flow<EntityTemplate>
    suspend fun getByUserIdTypeKeyword(userId: String?, type: String?, keyword: String?, includeEntities: Boolean?): Flow<EntityTemplate>
    suspend fun getByTypeKeyword(type: String?, keyword: String?, includeEntities: Boolean?): Flow<EntityTemplate>

}
